### DISCUSSION TITLE
OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a Python package or a valid path to a data directory.

### DISCUSSION THREAD

**Discussion by lakchchayam** [explosion/spaCy#8821] [Help: Installation] [answered]
![image](https://user-images.githubusercontent.com/49754403/127270438-4caf10cb-b056-43aa-b2f2-d1361e14aab8.png)

**Comment by polm** [comment, marked as answer]
You probably need to run `spacy download en_core_web_sm`. 

You post has a lot of issues. Here are some things that makes it easier for us to help you:

1. Explain what you're trying to do, and how you got this error. Don't just post an error message.
2. Say what version of spaCy you're using, and if possible share the output of `spacy info`.
3. Don't post code as screenshots, copy/paste it so we can search the text. Also screenshots may be small or blurry and hard to read.

I'm sorry to give a big list of things on your first issue here, but following the steps above really does help us figure out your problem and how to fix it faster.

**Reply by lakchchayam** [reply]
I'm trying to do topic modelling but tried everything with 'en-core-web-sm' it is showing error everytime 

```
(spa) D:\Universite de Sherbrooke\testing\spacytests>spacy info

============================== Info about spaCy ==============================

spaCy version    3.0.7
Location         c:\users\lakch\appdata\local\programs\python\python39\lib\site-packages\spacy
Platform         Windows-10-10.0.19043-SP0
Python version   3.9.5
Pipelines        en_core_web_sm (3.0.0)



---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-9-771817c5e36c> in <module>
      1 import spacy
----> 2 nlp = spacy.load('en_core_web_sm')
      3 def lemmatization(texts, allowed_postags=["NOUN", "ADJ", "VERB", "ADV"]):
      4     nlp = spacy.load("en_core_web_sm", disable=["parser", "ner"])
      5     texts_out = []

~\anaconda3\lib\site-packages\spacy\__init__.py in load(name, vocab, disable, exclude, config)
     49     RETURNS (Language): The loaded nlp object.
     50     """
---> 51     return util.load_model(
     52         name, vocab=vocab, disable=disable, exclude=exclude, config=config
     53     )

~\anaconda3\lib\site-packages\spacy\util.py in load_model(name, vocab, disable, exclude, config)
    329     if name in OLD_MODEL_SHORTCUTS:
    330         raise IOError(Errors.E941.format(name=name, full=OLD_MODEL_SHORTCUTS[name]))
--> 331     raise IOError(Errors.E050.format(name=name))
    332 
    333 

OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a Python package or a valid path to a data directory.
​

**Reply by lakchchayam** [reply]
tried doing normally and with virtual environment but nothing worked

**Reply by lakchchayam** [reply]
![image](https://user-images.githubusercontent.com/49754403/127288987-d2485017-78de-4e1d-8076-29156a913627.png)

**Reply by lakchchayam** [reply]
Can't even run your examples

**Reply by polm** [reply]
Please don't paste screenshots of your error, it's unhelpful.

Thank you for posting the logs, I think I know what's wrong.

Your `spacy info` output info gives this path, which is using vanilla Python:

    c:\users\lakch\appdata\local\programs\python\python39\

But when you get the error you can see that spaCy is installed in a different path, with Anaconda:

    ~\anaconda3\lib\site-packages\spacy

I was able to reproduce your error by installing spaCy and downloading the model with vanilla Python and then installing spaCy separately in Anaconda. Anaconda can't look up the model installed with vanilla Python. 

The way Anaconda and Python interact can be confusing and I'm not sure how you're running Python (IDE/Anaconda shell/etc), but can you try following [the Quickstart](https://spacy.io/usage) from scratch? Some of the ways it does things look unremarkable but are designed to avoid issues like this. For example, I usually use `spacy download` on Linux without Anaconda but maybe that's not the right thing to do in your environment (maybe the `python -m spacy download` part is important).

Also, for the record, spaCy doesn't have any topic modelling features, though there's probably a third-party plugin for them.

**Reply by lakchchayam** [reply]
In this I have tried creating virtual environment and python -m spacy download but still same error exist

```
Microsoft Windows [Version 10.0.19043.1110]
(c) Microsoft Corporation. All rights reserved.

D:\Universite de Sherbrooke\testing\testforencorewebsm>conda create -n encore
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\lakch\anaconda3\envs\encore



Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate encore
#
# To deactivate an active environment, use
#
#     $ conda deactivate


D:\Universite de Sherbrooke\testing\testforencorewebsm>conda activate encore

(encore) D:\Universite de Sherbrooke\testing\testforencorewebsm>conda install spacy
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: C:\Users\lakch\anaconda3\envs\encore

  added / updated specs:
    - spacy


The following NEW packages will be INSTALLED:

  attrs              pkgs/main/noarch::attrs-21.2.0-pyhd3eb1b0_0
  blas               pkgs/main/win-64::blas-1.0-mkl
  brotlipy           pkgs/main/win-64::brotlipy-0.7.0-py39h2bbff1b_1003
  ca-certificates    pkgs/main/win-64::ca-certificates-2021.7.5-haa95532_1
  catalogue          pkgs/main/noarch::catalogue-1.0.0-pyhd3eb1b0_0
  certifi            pkgs/main/win-64::certifi-2021.5.30-py39haa95532_0
  cffi               pkgs/main/win-64::cffi-1.14.6-py39h2bbff1b_0
  chardet            pkgs/main/win-64::chardet-4.0.0-py39haa95532_1003
  cryptography       pkgs/main/win-64::cryptography-3.4.7-py39h71e12ea_0
  cymem              pkgs/main/win-64::cymem-2.0.5-py39hd77b12b_0
  cython-blis        pkgs/main/win-64::cython-blis-0.7.4-py39h2bbff1b_1
  idna               pkgs/main/noarch::idna-2.10-pyhd3eb1b0_0
  intel-openmp       pkgs/main/win-64::intel-openmp-2021.3.0-haa95532_3372
  jsonschema         pkgs/main/win-64::jsonschema-3.0.2-py39haa95532_0
  mkl                pkgs/main/win-64::mkl-2021.3.0-haa95532_524
  mkl-service        pkgs/main/win-64::mkl-service-2.4.0-py39h2bbff1b_0
  mkl_fft            pkgs/main/win-64::mkl_fft-1.3.0-py39h277e83a_2
  mkl_random         pkgs/main/win-64::mkl_random-1.2.2-py39hf11a4ad_0
  murmurhash         pkgs/main/win-64::murmurhash-1.0.5-py39hd77b12b_0
  numpy              pkgs/main/win-64::numpy-1.20.3-py39ha4e8547_0
  numpy-base         pkgs/main/win-64::numpy-base-1.20.3-py39hc2deb75_0
  openssl            pkgs/main/win-64::openssl-1.1.1k-h2bbff1b_0
  pip                pkgs/main/win-64::pip-21.1.3-py39haa95532_0
  plac               pkgs/main/win-64::plac-1.1.0-py39haa95532_0
  preshed            pkgs/main/win-64::preshed-3.0.5-py39h6c2663c_4
  pycparser          pkgs/main/noarch::pycparser-2.20-py_2
  pyopenssl          pkgs/main/noarch::pyopenssl-20.0.1-pyhd3eb1b0_1
  pyrsistent         pkgs/main/win-64::pyrsistent-0.18.0-py39h2bbff1b_0
  pysocks            pkgs/main/win-64::pysocks-1.7.1-py39haa95532_0
  python             pkgs/main/win-64::python-3.9.5-h6244533_3
  requests           pkgs/main/noarch::requests-2.25.1-pyhd3eb1b0_0
  setuptools         pkgs/main/win-64::setuptools-52.0.0-py39haa95532_0
  six                pkgs/main/noarch::six-1.16.0-pyhd3eb1b0_0
  spacy              pkgs/main/win-64::spacy-2.3.5-py39h59b6b97_0
  sqlite             pkgs/main/win-64::sqlite-3.36.0-h2bbff1b_0
  srsly              pkgs/main/win-64::srsly-1.0.5-py39hd77b12b_0
  thinc              pkgs/main/win-64::thinc-7.4.5-py39h757b272_0
  tqdm               pkgs/main/noarch::tqdm-4.61.2-pyhd3eb1b0_1
  tzdata             pkgs/main/noarch::tzdata-2021a-h52ac0ba_0
  urllib3            pkgs/main/noarch::urllib3-1.26.6-pyhd3eb1b0_1
  vc                 pkgs/main/win-64::vc-14.2-h21ff451_1
  vs2015_runtime     pkgs/main/win-64::vs2015_runtime-14.27.29016-h5e58377_2
  wasabi             pkgs/main/noarch::wasabi-0.8.2-pyhd3eb1b0_0
  wheel              pkgs/main/noarch::wheel-0.36.2-pyhd3eb1b0_0
  win_inet_pton      pkgs/main/win-64::win_inet_pton-1.1.0-py39haa95532_0
  wincertstore       pkgs/main/win-64::wincertstore-0.2-py39h2bbff1b_0


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done

(encore) D:\Universite de Sherbrooke\testing\testforencorewebsm>python -m spacy download en_core_web_sm
⚠ Skipping model package dependencies and setting `--no-deps`. You don't seem to
have the spaCy package itself installed (maybe because you've built from
source?), so installing the model dependencies would cause spaCy to be
downloaded, which probably isn't what you want. If the model package has other
dependencies, you'll have to install them manually.
Collecting en_core_web_sm==2.3.1
  Downloading https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.3.1/en_core_web_sm-2.3.1.tar.gz (12.0 MB)
     |████████████████████████████████| 12.0 MB 3.3 MB/s
Building wheels for collected packages: en-core-web-sm
  Building wheel for en-core-web-sm (setup.py) ... done
  Created wheel for en-core-web-sm: filename=en_core_web_sm-2.3.1-py3-none-any.whl size=12047106 sha256=caf6a40b0c0d4bf78dfd14fc3f9198b8056909d094c9b9087179b259301f08a3
  Stored in directory: C:\Users\lakch\AppData\Local\Temp\pip-ephem-wheel-cache-6dcnntdx\wheels\19\d6\1c\5484b95647df5d7afaf74abde458c66c1cd427e69e801fe826
Successfully built en-core-web-sm
Installing collected packages: en-core-web-sm
Successfully installed en-core-web-sm-2.3.1
✔ Download and installation successful
You can now load the model via spacy.load('en_core_web_sm')

(encore) D:\Universite de Sherbrooke\testing\testforencorewebsm>
```

**Reply by lakchchayam** [reply]
Still same issue after creating environment 

https://stackoverflow.com/questions/66367475/oserror-e050-cant-find-model-en-core-web-sm-it-doesnt-seem-to-be-a-short

**Reply by polm** [reply]
You didn't follow the quickstart instructions. These are the instructions for Anaconda on Windows:

```
conda install -c conda-forge spacy
python -m spacy download en_core_web_sm
```

Because you used `conda install spacy`, conda installs v2.3.5, which is old. We've seen this before and it's not really clear why it happens, it seems conda avoids upgrading something else in the default env and uses the latest compatible spaCy version. 

Could you please try following the instructions?

**Reply by lakchchayam** [reply]
Ok lets try this. Do I follow Quickstart steps on virtual env or globally?

**Reply by polm** [reply]
Quickstart should work in a virtualenv or globally, but it's always better to use a clean virtualenv. Using a fresh env with conda rather than the base env also sometimes resolves issues, as described in the issue you linked to.

**Reply by lakchchayam** [reply]
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
<ipython-input-3-1c080c493f24> in <module>
      1 import spacy
      2 
----> 3 nlp = spacy.load("en_core_web_sm")
      4 doc = nlp("Apple is looking at buying U.K. startup for $1 billion")
      5 

~\anaconda3\lib\site-packages\spacy\__init__.py in load(name, vocab, disable, exclude, config)
     49     RETURNS (Language): The loaded nlp object.
     50     """
---> 51     return util.load_model(
     52         name, vocab=vocab, disable=disable, exclude=exclude, config=config
     53     )

~\anaconda3\lib\site-packages\spacy\util.py in load_model(name, vocab, disable, exclude, config)
    329     if name in OLD_MODEL_SHORTCUTS:
    330         raise IOError(Errors.E941.format(name=name, full=OLD_MODEL_SHORTCUTS[name]))
--> 331     raise IOError(Errors.E050.format(name=name))
    332 
    333 

OSError: [E050] Can't find model 'en_core_web_sm'. It doesn't seem to be a Python package or a valid path to a data directory.

**Reply by lakchchayam** [reply]
Error finally solved because encore library is not supported by latest version of spacy that is 3.0.1 but when I tried with pip install 
spacy == 3.0.0 it succesfully compiled

**Reply by polm** [reply]
Uh, you posted a lot of logs without comment or explanation and then deleted some of them, so I'm not entirely sure what happened, but glad you got it working.

At an earlier point you had an error about the model v3.0.0 and spaCy v3.1.1 not being compatible. That can happen if you download the model and then upgrade spaCy. After upgrading you need to download the model again to get the latest version.

Also note the latest version of spaCy is 3.1.1. 

Also, some of your deleted comments referred to LDA, so I'll mention again that spaCy has no features for topic modelling or LDA.

**Reply by lakchchayam** [reply]
Do you know the latest version of spacyv3.0.1 is not compatible with encore library
